Traffic Ops unit tests: TestGetMonitoringJSON sometimes fails

<!--
************ STOP!! ************
If this issue identifies a security vulnerability, DO NOT submit it! Instead, contact
the Apache Software Foundation Security Team at security@trafficcontrol.apache.org and follow the
guidelines at https://www.apache.org/security/ regarding vulnerability disclosure.

- For *SUPPORT QUESTIONS*, use the
[Traffic Control slack channels](https://traffic-control-cdn.slack.com) or [Traffic Control mailing lists](http://trafficcontrol.apache.org/mailing_lists/).
- Before submitting, please **SEARCH GITHUB** for a similar issue or PR. -->

## I'm submitting a ...
<!-- delete all those that don't apply -->
<!--- security vulnerability (STOP!! - see above)-->
-  bug report

## Traffic Control components affected ...
<!-- delete all those that don't apply -->
-  Traffic Ops

## Current behavior:
<!-- Describe how the bug manifests / how the current features are insufficient. -->
`github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/monitoring.TestGetMonitoringJSON` usually passes but sometimes fails:
```shell script
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc000393220 ServiceAddress:true} {Address:2020::4 Gateway:0xc000393230 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000393240 ServiceAddress:false} {Address:2021::4 Gateway:0xc000393250 ServiceAddress:false}] MaxBandwidth:0xc0003967a0 Monitor:true MTU:0xc0003967a8 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000393260 ServiceAddress:true} {Address:2021::4 Gateway:0xc000393270 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000393280 ServiceAddress:false} {Address:2022::4 Gateway:0xc000393290 ServiceAddress:false}] MaxBandwidth:0xc0003967b0 Monitor:false MTU:0xc0003967b8 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000393260 ServiceAddress:true} {Address:2021::4 Gateway:0xc000393270 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000393280 ServiceAddress:false} {Address:2022::4 Gateway:0xc000393290 ServiceAddress:false}] MaxBandwidth:0xc0003967b0 Monitor:false MTU:0xc0003967b8 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc000393220 ServiceAddress:true} {Address:2020::4 Gateway:0xc000393230 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000393240 ServiceAddress:false} {Address:2021::4 Gateway:0xc000393250 ServiceAddress:false}] MaxBandwidth:0xc0003967a0 Monitor:true MTU:0xc0003967a8 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
```

## Expected / new behavior:
<!-- Describe what the behavior would be without the bug / how the feature would improve Traffic Control -->
All unit tests should always pass

## Minimal reproduction of the problem with instructions:
<!--
If the current behavior is a bug or you can illustrate your feature request better with an example,
please provide the *STEPS TO REPRODUCE* and include the applicable TC version.
-->
```shell script
cd traffic_ops/traffic_ops_golang/monitoring
go test -c
failures=0
for ((i = 0; i < 100; i++)); do
  ./monitoring.test || (( ++failures ))
done;
echo "Failed ${failures} times out of ${i}"
```

## Anything else:
<!-- e.g. stacktraces, related issues, suggestions how to fix -->
If running the unit tests multiple times, the number of failures varies. <details><summary>Example output from running the tests 100 times as shown above (click to expand):</summary>

```golang
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc000293800 ServiceAddress:true} {Address:2020::4 Gateway:0xc000293810 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000293820 ServiceAddress:false} {Address:2021::4 Gateway:0xc000293830 ServiceAddress:false}] MaxBandwidth:0xc00029ab50 Monitor:true MTU:0xc00029ab58 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000293840 ServiceAddress:true} {Address:2021::4 Gateway:0xc000293850 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000293860 ServiceAddress:false} {Address:2022::4 Gateway:0xc000293870 ServiceAddress:false}] MaxBandwidth:0xc00029ab60 Monitor:false MTU:0xc00029ab68 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000293840 ServiceAddress:true} {Address:2021::4 Gateway:0xc000293850 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000293860 ServiceAddress:false} {Address:2022::4 Gateway:0xc000293870 ServiceAddress:false}] MaxBandwidth:0xc00029ab60 Monitor:false MTU:0xc00029ab68 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc000293800 ServiceAddress:true} {Address:2020::4 Gateway:0xc000293810 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000293820 ServiceAddress:false} {Address:2021::4 Gateway:0xc000293830 ServiceAddress:false}] MaxBandwidth:0xc00029ab50 Monitor:true MTU:0xc00029ab58 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002b7260 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002b7270 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002b7280 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002b7290 ServiceAddress:false}] MaxBandwidth:0xc0002cc6b0 Monitor:true MTU:0xc0002cc6b8 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002b72a0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002b72b0 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002b72c0 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002b72d0 ServiceAddress:false}] MaxBandwidth:0xc0002cc6c0 Monitor:false MTU:0xc0002cc6c8 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002b72a0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002b72b0 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002b72c0 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002b72d0 ServiceAddress:false}] MaxBandwidth:0xc0002cc6c0 Monitor:false MTU:0xc0002cc6c8 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002b7260 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002b7270 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002b7280 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002b7290 ServiceAddress:false}] MaxBandwidth:0xc0002cc6b0 Monitor:true MTU:0xc0002cc6b8 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0001137e0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0001137f0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000113800 ServiceAddress:false} {Address:2021::4 Gateway:0xc000113810 ServiceAddress:false}] MaxBandwidth:0xc000116b40 Monitor:true MTU:0xc000116b48 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000113820 ServiceAddress:true} {Address:2021::4 Gateway:0xc000113830 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000113840 ServiceAddress:false} {Address:2022::4 Gateway:0xc000113850 ServiceAddress:false}] MaxBandwidth:0xc000116b50 Monitor:false MTU:0xc000116b58 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000113820 ServiceAddress:true} {Address:2021::4 Gateway:0xc000113830 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000113840 ServiceAddress:false} {Address:2022::4 Gateway:0xc000113850 ServiceAddress:false}] MaxBandwidth:0xc000116b50 Monitor:false MTU:0xc000116b58 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0001137e0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0001137f0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000113800 ServiceAddress:false} {Address:2021::4 Gateway:0xc000113810 ServiceAddress:false}] MaxBandwidth:0xc000116b40 Monitor:true MTU:0xc000116b48 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc000297880 ServiceAddress:true} {Address:2020::4 Gateway:0xc000297890 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002978a0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002978b0 ServiceAddress:false}] MaxBandwidth:0xc00029ab30 Monitor:true MTU:0xc00029ab38 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002978c0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002978d0 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002978e0 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002978f0 ServiceAddress:false}] MaxBandwidth:0xc00029ab40 Monitor:false MTU:0xc00029ab48 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002978c0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002978d0 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002978e0 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002978f0 ServiceAddress:false}] MaxBandwidth:0xc00029ab40 Monitor:false MTU:0xc00029ab48 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc000297880 ServiceAddress:true} {Address:2020::4 Gateway:0xc000297890 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002978a0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002978b0 ServiceAddress:false}] MaxBandwidth:0xc00029ab30 Monitor:true MTU:0xc00029ab38 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc000115820 ServiceAddress:true} {Address:2020::4 Gateway:0xc000115830 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000115840 ServiceAddress:false} {Address:2021::4 Gateway:0xc000115850 ServiceAddress:false}] MaxBandwidth:0xc00011ab30 Monitor:true MTU:0xc00011ab38 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000115860 ServiceAddress:true} {Address:2021::4 Gateway:0xc000115870 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000115880 ServiceAddress:false} {Address:2022::4 Gateway:0xc000115890 ServiceAddress:false}] MaxBandwidth:0xc00011ab40 Monitor:false MTU:0xc00011ab48 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000115860 ServiceAddress:true} {Address:2021::4 Gateway:0xc000115870 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000115880 ServiceAddress:false} {Address:2022::4 Gateway:0xc000115890 ServiceAddress:false}] MaxBandwidth:0xc00011ab40 Monitor:false MTU:0xc00011ab48 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc000115820 ServiceAddress:true} {Address:2020::4 Gateway:0xc000115830 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000115840 ServiceAddress:false} {Address:2021::4 Gateway:0xc000115850 ServiceAddress:false}] MaxBandwidth:0xc00011ab30 Monitor:true MTU:0xc00011ab38 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0003036c0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0003036d0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0003036e0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0003036f0 ServiceAddress:false}] MaxBandwidth:0xc000310b00 Monitor:true MTU:0xc000310b08 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000303700 ServiceAddress:true} {Address:2021::4 Gateway:0xc000303710 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000303720 ServiceAddress:false} {Address:2022::4 Gateway:0xc000303730 ServiceAddress:false}] MaxBandwidth:0xc000310b10 Monitor:false MTU:0xc000310b18 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000303700 ServiceAddress:true} {Address:2021::4 Gateway:0xc000303710 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000303720 ServiceAddress:false} {Address:2022::4 Gateway:0xc000303730 ServiceAddress:false}] MaxBandwidth:0xc000310b10 Monitor:false MTU:0xc000310b18 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0003036c0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0003036d0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0003036e0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0003036f0 ServiceAddress:false}] MaxBandwidth:0xc000310b00 Monitor:true MTU:0xc000310b08 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc000385840 ServiceAddress:true} {Address:2020::4 Gateway:0xc000385850 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000385860 ServiceAddress:false} {Address:2021::4 Gateway:0xc000385870 ServiceAddress:false}] MaxBandwidth:0xc00038ab30 Monitor:true MTU:0xc00038ab38 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000385880 ServiceAddress:true} {Address:2021::4 Gateway:0xc000385890 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0003858a0 ServiceAddress:false} {Address:2022::4 Gateway:0xc0003858b0 ServiceAddress:false}] MaxBandwidth:0xc00038ab40 Monitor:false MTU:0xc00038ab48 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000385880 ServiceAddress:true} {Address:2021::4 Gateway:0xc000385890 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0003858a0 ServiceAddress:false} {Address:2022::4 Gateway:0xc0003858b0 ServiceAddress:false}] MaxBandwidth:0xc00038ab40 Monitor:false MTU:0xc00038ab48 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc000385840 ServiceAddress:true} {Address:2020::4 Gateway:0xc000385850 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc000385860 ServiceAddress:false} {Address:2021::4 Gateway:0xc000385870 ServiceAddress:false}] MaxBandwidth:0xc00038ab30 Monitor:true MTU:0xc00038ab38 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0000218d0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0000218e0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0000218f0 ServiceAddress:false} {Address:2021::4 Gateway:0xc000021900 ServiceAddress:false}] MaxBandwidth:0xc000024c40 Monitor:true MTU:0xc000024c48 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000021910 ServiceAddress:true} {Address:2021::4 Gateway:0xc000021920 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000021930 ServiceAddress:false} {Address:2022::4 Gateway:0xc000021940 ServiceAddress:false}] MaxBandwidth:0xc000024c50 Monitor:false MTU:0xc000024c58 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000021910 ServiceAddress:true} {Address:2021::4 Gateway:0xc000021920 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000021930 ServiceAddress:false} {Address:2022::4 Gateway:0xc000021940 ServiceAddress:false}] MaxBandwidth:0xc000024c50 Monitor:false MTU:0xc000024c58 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0000218d0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0000218e0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0000218f0 ServiceAddress:false} {Address:2021::4 Gateway:0xc000021900 ServiceAddress:false}] MaxBandwidth:0xc000024c40 Monitor:true MTU:0xc000024c48 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002b51b0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002b51c0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002b51d0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002b51e0 ServiceAddress:false}] MaxBandwidth:0xc0002ca6a0 Monitor:true MTU:0xc0002ca6a8 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002b51f0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002b5200 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002b5210 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002b5220 ServiceAddress:false}] MaxBandwidth:0xc0002ca6b0 Monitor:false MTU:0xc0002ca6b8 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002b51f0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002b5200 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002b5210 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002b5220 ServiceAddress:false}] MaxBandwidth:0xc0002ca6b0 Monitor:false MTU:0xc0002ca6b8 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002b51b0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002b51c0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002b51d0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002b51e0 ServiceAddress:false}] MaxBandwidth:0xc0002ca6a0 Monitor:true MTU:0xc0002ca6a8 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002928d0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002928e0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002928f0 ServiceAddress:false} {Address:2021::4 Gateway:0xc000292900 ServiceAddress:false}] MaxBandwidth:0xc000298410 Monitor:true MTU:0xc000298418 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000292910 ServiceAddress:true} {Address:2021::4 Gateway:0xc000292920 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000292930 ServiceAddress:false} {Address:2022::4 Gateway:0xc000292940 ServiceAddress:false}] MaxBandwidth:0xc000298420 Monitor:false MTU:0xc000298428 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000292910 ServiceAddress:true} {Address:2021::4 Gateway:0xc000292920 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000292930 ServiceAddress:false} {Address:2022::4 Gateway:0xc000292940 ServiceAddress:false}] MaxBandwidth:0xc000298420 Monitor:false MTU:0xc000298428 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002928d0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002928e0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002928f0 ServiceAddress:false} {Address:2021::4 Gateway:0xc000292900 ServiceAddress:false}] MaxBandwidth:0xc000298410 Monitor:true MTU:0xc000298418 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002b71a0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002b71b0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002b71c0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002b71d0 ServiceAddress:false}] MaxBandwidth:0xc0002cc6a0 Monitor:true MTU:0xc0002cc6a8 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002b71e0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002b71f0 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002b7200 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002b7210 ServiceAddress:false}] MaxBandwidth:0xc0002cc6b0 Monitor:false MTU:0xc0002cc6b8 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc0002b71e0 ServiceAddress:true} {Address:2021::4 Gateway:0xc0002b71f0 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc0002b7200 ServiceAddress:false} {Address:2022::4 Gateway:0xc0002b7210 ServiceAddress:false}] MaxBandwidth:0xc0002cc6b0 Monitor:false MTU:0xc0002cc6b8 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0002b71a0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0002b71b0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0002b71c0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0002b71d0 ServiceAddress:false}] MaxBandwidth:0xc0002cc6a0 Monitor:true MTU:0xc0002cc6a8 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMonitoringJSON (0.00s)
    monitoring_test.go:544: GetMonitoringJSON expected TrafficServers: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:5.6.7.8 Gateway:0xc0003911c0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0003911d0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0003911e0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0003911f0 ServiceAddress:false}] MaxBandwidth:0xc0003947b0 Monitor:true MTU:0xc0003947b8 Name:test1} {IPAddresses:[{Address:6.7.8.9 Gateway:0xc000391200 ServiceAddress:true} {Address:2021::4 Gateway:0xc000391210 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000391220 ServiceAddress:false} {Address:2022::4 Gateway:0xc000391230 ServiceAddress:false}] MaxBandwidth:0xc0003947c0 Monitor:false MTU:0xc0003947c8 Name:test2}] Type:EDGE HashID:cacheHash}] actual: [{CommonServerProperties:{Profile:cacheProfile Status:cacheStatus Port:8081 Cachegroup:cacheCachegroup HostName:cacheHost FQDN:cacheFqdn.me} Interfaces:[{IPAddresses:[{Address:6.7.8.9 Gateway:0xc000391200 ServiceAddress:true} {Address:2021::4 Gateway:0xc000391210 ServiceAddress:true} {Address:6.6.7.9 Gateway:0xc000391220 ServiceAddress:false} {Address:2022::4 Gateway:0xc000391230 ServiceAddress:false}] MaxBandwidth:0xc0003947c0 Monitor:false MTU:0xc0003947c8 Name:test2} {IPAddresses:[{Address:5.6.7.8 Gateway:0xc0003911c0 ServiceAddress:true} {Address:2020::4 Gateway:0xc0003911d0 ServiceAddress:true} {Address:5.6.7.9 Gateway:0xc0003911e0 ServiceAddress:false} {Address:2021::4 Gateway:0xc0003911f0 ServiceAddress:false}] MaxBandwidth:0xc0003947b0 Monitor:true MTU:0xc0003947b8 Name:test1}] Type:EDGE HashID:cacheHash}]
FAIL
PASS
Failed 12 times out of 100
```
</details>

<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

--- a/traffic_ops/traffic_ops_golang/monitoring/monitoring_test.go
+++ b/traffic_ops/traffic_ops_golang/monitoring/monitoring_test.go
@@ -392,6 +392,20 @@ func TestGetConfig(t *testing.T) {
 	}
 }
 
+func sortByInterfaceName(servers []Cache) {
+	for _, cache := range servers {
+		sort.SliceStable(cache.Interfaces, func(i, j int) bool {
+			interface1, interface2 := cache.Interfaces[i], cache.Interfaces[j]
+			switch {
+			case interface1.Name != interface2.Name:
+				return interface1.Name < interface2.Name
+			default:
+				return interface1.Name < interface2.Name
+			}
+		})
+	}
+}
+
 func TestGetMonitoringJSON(t *testing.T) {
 	resp := MonitoringResponse{}
 	mockDB, mock, err := sqlmock.New()
@@ -528,9 +542,10 @@ func TestGetMonitoringJSON(t *testing.T) {
 	if err != nil {
 		t.Errorf("GetMonitoringJSON expected: nil error, actual: %v", err)
 	}
-
 	resp.Response.TrafficServers = sortCaches(resp.Response.TrafficServers)
 	sqlResp.TrafficServers = sortCaches(sqlResp.TrafficServers)
+	sortByInterfaceName(resp.Response.TrafficServers)
+	sortByInterfaceName(sqlResp.TrafficServers)
 	resp.Response.TrafficMonitors = sortMonitors(resp.Response.TrafficMonitors)
 	sqlResp.TrafficMonitors = sortMonitors(sqlResp.TrafficMonitors)
 	resp.Response.Cachegroups = sortCachegroups(resp.Response.Cachegroups)

--- a/traffic_ops/traffic_ops_golang/monitoring/monitoring_test.go
+++ b/traffic_ops/traffic_ops_golang/monitoring/monitoring_test.go
@@ -544,8 +544,6 @@ func TestGetMonitoringJSON(t *testing.T) {
 	}
 	resp.Response.TrafficServers = sortCaches(resp.Response.TrafficServers)
 	sqlResp.TrafficServers = sortCaches(sqlResp.TrafficServers)
-	sortByInterfaceName(resp.Response.TrafficServers)
-	sortByInterfaceName(sqlResp.TrafficServers)
 	resp.Response.TrafficMonitors = sortMonitors(resp.Response.TrafficMonitors)
 	sqlResp.TrafficMonitors = sortMonitors(sqlResp.TrafficMonitors)
 	resp.Response.Cachegroups = sortCachegroups(resp.Response.Cachegroups)
@@ -623,6 +621,7 @@ func (s SortableCaches) Less(i, j int) bool {
 
 func sortCaches(p []Cache) []Cache {
 	sort.Sort(SortableCaches(p))
+	sortByInterfaceName(p)
 	return p
 }
 

--- a/traffic_ops/traffic_ops_golang/monitoring/monitoring_test.go
+++ b/traffic_ops/traffic_ops_golang/monitoring/monitoring_test.go
@@ -392,20 +392,6 @@ func TestGetConfig(t *testing.T) {
 	}
 }
 
-func sortByInterfaceName(servers []Cache) {
-	for _, cache := range servers {
-		sort.SliceStable(cache.Interfaces, func(i, j int) bool {
-			interface1, interface2 := cache.Interfaces[i], cache.Interfaces[j]
-			switch {
-			case interface1.Name != interface2.Name:
-				return interface1.Name < interface2.Name
-			default:
-				return interface1.Name < interface2.Name
-			}
-		})
-	}
-}
-
 func TestGetMonitoringJSON(t *testing.T) {
 	resp := MonitoringResponse{}
 	mockDB, mock, err := sqlmock.New()
@@ -542,6 +528,12 @@ func TestGetMonitoringJSON(t *testing.T) {
 	if err != nil {
 		t.Errorf("GetMonitoringJSON expected: nil error, actual: %v", err)
 	}
+	for _, cache := range resp.Response.TrafficServers {
+		cache.Interfaces = sortInterfaces(cache.Interfaces)
+	}
+	for _, cache := range sqlResp.TrafficServers {
+		cache.Interfaces = sortInterfaces(cache.Interfaces)
+	}
 	resp.Response.TrafficServers = sortCaches(resp.Response.TrafficServers)
 	sqlResp.TrafficServers = sortCaches(sqlResp.TrafficServers)
 	resp.Response.TrafficMonitors = sortMonitors(resp.Response.TrafficMonitors)
@@ -619,9 +611,25 @@ func (s SortableCaches) Less(i, j int) bool {
 	return s[i].HostName < s[j].HostName
 }
 
+type SortableInterfaces []tc.ServerInterfaceInfo
+
+func (s SortableInterfaces) Len() int {
+	return len(s)
+}
+func (s SortableInterfaces) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+func (s SortableInterfaces) Less(i, j int) bool {
+	return s[i].Name < s[j].Name
+}
+
+func sortInterfaces(i []tc.ServerInterfaceInfo) []tc.ServerInterfaceInfo {
+	sort.Sort(SortableInterfaces(i))
+	return i
+}
+
 func sortCaches(p []Cache) []Cache {
 	sort.Sort(SortableCaches(p))
-	sortByInterfaceName(p)
 	return p
 }
 

